extensions/binding/org.eclipse.smarthome.binding.iota.test/src/test/java/org/eclipse/smarthome/binding/iota/IotaBridgeHandlerTest.java

```diff
@@ -0,0 +1,94 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http:www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.iota;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.binding.iota.handler.IotaBridgeHandler;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerCallback;
+import org.eclipse.smarthome.core.thing.binding.builder.BridgeBuilder;
+import org.eclipse.smarthome.io.iota.utils.IotaUtils;
+import org.eclipse.smarthome.test.java.JavaTest;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * The {@link IotaBridgeHandlerTest} provides test cases for the {@link IotaBridgeHandler}. The tests provide mocks for
+ * supporting entities using Mockito.
+ *
+ * @author Theo Giovanna - Initial contribution
+ */
+public class IotaBridgeHandlerTest extends JavaTest {
+
+    private MockIotaBridgeHandler iotaBridgeHandler;
+    private Map<String, Object> bridgeProperties;
+    private Bridge bridge;
+
+    @Before
+    public void setUp() {
+        initializeBridge();
+    }
+
+    @Test
+    public void bridgeShouldInitialize() {
+        waitForAssert(() -> assertEquals(ThingStatus.ONLINE, bridge.getStatus()));
+    }
+
+    @Test
+    public void checkDefaultApiIsOnline() {
+        assertEquals(iotaBridgeHandler.getUtils().checkAPI(), true);
+    }
+
+    private void initializeBridge() {
+        bridgeProperties = new HashMap<>();
+        bridge = BridgeBuilder.create(new ThingTypeUID("iota", "test-bridge"), "testbridge").withLabel("Test Bridge")
+                .withConfiguration(new Configuration(bridgeProperties)).build();
+        iotaBridgeHandler = new IotaBridgeHandlerTest().new MockIotaBridgeHandler(bridge);
+        bridge.setHandler(iotaBridgeHandler);
+        ThingHandlerCallback bridgeHandler = mock(ThingHandlerCallback.class);
+        doAnswer(answer -> {
+            ((Thing) answer.getArgument(0)).setStatusInfo(answer.getArgument(1));
+            return null;
+        }).when(bridgeHandler).statusUpdated(any(), any());
+        iotaBridgeHandler.setCallback(bridgeHandler);
+        iotaBridgeHandler.initialize();
+    }
+
+    class MockIotaBridgeHandler extends IotaBridgeHandler {
+
+        MockIotaBridgeHandler(Bridge bridge) {
+            super(bridge);
+        }
+
+        @Override
+        public void initialize() {
+            this.setProtocol("https");
+            this.setHost("nodes.testnet.iota.org");
```

Comment:
The bridge really only contains an `IotaAPI` instance that acts as the communication bridge with the Tangle. I don't see how I can test it if parameters are not set explicitly, since the HandlerCallback returns ONLINE only when `utils.checkAPI()` returns true, which call needs a defined node. Even if I set a config map with the parameters, we still rely on the node being online (otherwise the test fails, which I agree isn't a good thing). So not sure what to do here.

Reply:
The properties set here (protocol, host, port) are used inside the IOTAUtil class which is the actual communication client. For the test it would be nice to have a different version of IOTAUtils which does not contact a live node but just mocks the behaviour and returns status/values you want it to during the test.
Right now IOTAUtils gets instantiated inside the thing handler, so you can not exchange it easily in the test. To achieve this you might just give the IOTAUtils instance from the handler factory to the thing handler. With this construct you can also give a different instance to the thing handler in the test. To make it more easy to create a mock instance from IOTAUtils you should make the class an interface and rename the current IOTAUtils to DefaultIOTAUtils or IOTAUtilsImpl.
The mocked version in the test may either be a new class you implement just for the sake of the test or you create one from Mockito with `iotaClient = mock(IOTAUtils.class)` and configure its behaviour by `when(iotaClient.xxxx()).thenReturn(....)`.

Reply:
On it.